src/service.rs

```diff
@@ -29,6 +30,40 @@ use hyper::body::Bytes;
 
 use tracing::{debug, error};
 
+
+use hyper::StatusCode;
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+
+
+use std::path::PathBuf;
+
+async fn serve_file(req: Request) -> S3Result<Response> {
```

Comment:
The over-engineer-er in me wants to integrate static-web-server here instead of doing an ad-hoc file server like this. I also think we should build the frontend code out of an actual nodejs project with minification and stuff.

The first part I can do very easily later, I already did it once for WinterJS.

The second part, we may need to do right now, because of the entire prompting for credentials stuff.

Maybe we can keep the FE part out for now and just merge in as much as is needed for the S3 serving part to work, and add the FE later on?